### PR TITLE
[16.0][FIX] event_sale: use sudo to write on registrations

### PR DESCRIPTION
**Pull request by jdoutreloux**
If the user doesn't have rights to registrations and tries to change the partner on a SO, this will raise an access error even if there's no registrations linked to the SO.
We use sudo to be coherent with what is done in _update_registrations.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by tde-banana-odoo** [comment]
Romine !

**Comment by tivisse** [comment]
RRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRomiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiine !

File: addons/event_sale/models/sale_order.py [hunk 1]
```diff
@@ -14,8 +14,8 @@ def write(self, vals):
         only. """
         result = super(SaleOrder, self).write(vals)
         if vals.get('partner_id'):
-            registrations_toupdate = self.env['event.registration'].search([('sale_order_id', 'in', self.ids)])
-            registrations_toupdate.write({'partner_id': vals['partner_id']})
+            registrations_toupdate = self.sudo().env['event.registration'].search([('sale_order_id', 'in', self.ids)])
```
Comment (tde-banana-odoo) [inline addons/event_sale/models/sale_order.py:17]: Seems legit, could refer to your doppelganger's commit https://github.com/odoo/odoo/commit/19878d011f8abdc737b7b702c4c79632670d4b78 that trigerred https://github.com/odoo/odoo/commit/fb17edd0b0d5c956f6991feabfa704566fa0a11b used in '_update_registrations'.

Side note: isn't it valid for 15 ? Didn't check the code :)

Comment (jdoutreloux) [reply addons/event_sale/models/sale_order.py:17]: @tde-banana-odoo  I've added a PR for 15.0
 https://github.com/odoo/odoo/pull/151506

File: addons/event_sale/models/sale_order.py [hunk 2]
```diff
@@ -14,8 +14,8 @@ def write(self, vals):
...
+            registrations_toupdate.sudo().write({'partner_id': vals['partner_id']})
```
[the remark this answers is not in the document]
Comment (jdoutreloux) [reply addons/event_sale/models/sale_order.py]: Fixed :)

**Comment by tde-banana-odoo** [comment]
Will be done in 15 at https://github.com/odoo/odoo/pull/151506 and forward ported.

**Pull request closed without merging**